lib/rust/ensogl/core/src/display/object/instance.rs

```diff
@@ -3058,20 +3059,29 @@ where
 
 impl Model {
     /// Reset size of this display object to values that can be computed statically. This is, use
-    /// the size property if it is set to pixels or to percent of the parent size. Otherwise, set
-    /// the size to zero.
+    /// the size property if it is set to pixels or to percent of the parent size. If the size is
+    /// set to hug and will be chainged in next layout reftesh, the size is set to 0.0. Otherwise,
+    /// the size does not change.
     fn reset_size_to_static_values<Dim>(&self, x: Dim, parent_size: f32)
     where Dim: ResolutionDim {
         let size = match self.layout.size.get_dim(x) {
             Size::Fixed(unit) => unit.resolve_const_and_percent(parent_size).unwrap_or(0.0),
-            Size::Hug => 0.0,
+            Size::Hug
+                if self.dirty.transformation.check()
+                    || self.dirty.modified_children.check_all()
+                    || self.dirty.removed_children.check_all() =>
+                0.0,
```

Comment:
Please check if it is possible to minimize the amount of checks, as we agreed via phone.